From: Paul McEnery <pmcenery@gmail.com>
Subject: Adds soname to libgalinette library

--- a/libgalinette/CMakeLists.txt
+++ b/libgalinette/CMakeLists.txt
@@ -1,7 +1,7 @@
 # Include & build
 include_directories   (.)
 add_library           (libgalinette SHARED device.c error.c firmware.c flash.c libgalinette.h libgalinette-private.h)
-set_target_properties (libgalinette PROPERTIES PREFIX "")
+set_target_properties (libgalinette PROPERTIES PREFIX "" SOVERSION 0)
 
 # Install
 install (TARGETS libgalinette LIBRARY DESTINATION ${libdir})
